boot/bootutil/src/encrypted.c

```diff
@@ -305,7 +305,7 @@ boot_enc_load(const struct image_header *hdr, const struct flash_area *fap,
 int
 boot_enc_valid(const struct flash_area *fap)
 {
-    return enc_state[fap->fa_id - FLASH_AREA_IMAGE_0].valid;
```

Comment:
This PR makes me aware of that this mathematics might be not compatible with zephyr flash_area numeration.

Reply:
I think it would be much better to pass the slot number instead of the flash area.

Reply:
If the values are not sequential for Zephyr, maybe it would be better to change it to something like:
```
int
boot_enc_valid(const struct flash_area *fap)
{
    switch (fap->fa_id) {                                                       
    case FLASH_AREA_IMAGE_0:                                                    
        return enc_state[0].valid;                                              
    case FLASH_AREA_IMAGE_1:                                                    
        return enc_state[1].valid;                                              
    default:                                                                    
        assert(0);
        return 0;                                                               
    }
}
```
But on another PR please!

Reply:
Also checking that other `enc_state` accesses do not use the same math.

Reply:
will prepare the patch